verifiers/internal/gcb/provenance.go

```diff
@@ -0,0 +1,308 @@
+package gcb
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsa01 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.1"
+	dsselib "github.com/secure-systems-lab/go-securesystemslib/dsse"
+
+	serrors "github.com/slsa-framework/slsa-verifier/errors"
+	"github.com/slsa-framework/slsa-verifier/options"
+	"github.com/slsa-framework/slsa-verifier/verifiers/internal/gcb/keys"
+)
+
+var GCBBuilderIDs = []string{"https://cloudbuild.googleapis.com/GoogleHostedWorker@v0.2"}
+
+type v01IntotoStatement struct {
+	intoto.StatementHeader
+	Predicate slsa01.ProvenancePredicate `json:"predicate"`
+}
+
+type gloudProvenance struct {
+	ImageSummary struct {
+		Digest              string `json:"digest"`
+		FullQualifiedDigest string `json:"fully_qualified_digest"`
+		Registry            string `json:"registry"`
+		Repsitory           string `json:"repository"`
+	} `json:"image_summary"`
+	ProvenanceSummary struct {
+		Provenance []struct {
+			Build struct {
+				// TODO: this is untrusted, we should remove it.
+				// IntotoStatement v01IntotoStatement `json:"intotoStatement"`
+			} `json:"build"`
+			Kind        string           `json:"kind"`
+			ResourceUri string           `json:"resourceUri"`
+			Envelope    dsselib.Envelope `json:"envelope"`
+		} `json:"provenance"`
+	} `json:"provenance_summary"`
+}
+
+type GCBProvenance struct {
+	gcloudProv                    *gloudProvenance
+	verifiedIntotoStatementStruct *v01IntotoStatement
+	// Note: may need to support envelope only.
```

Comment:
supporting just the predicate too might make it clear that this is already verified

Reply:
you're suggesting not adding the envelope, correct?
(I think the note is outdated)

Reply:
I removed the comment. PTAL